lib/verk/workers_manager.ex

```diff
@@ -99,7 +99,15 @@ defmodule Verk.WorkersManager do
     if workers != 0 do
       case QueueManager.dequeue(state.queue_manager_name, workers) do
         jobs when is_list(jobs) ->
-          for job <- jobs, do: job |> Job.decode! |> start_job(state)
+          Enum.map(jobs, fn job ->
+            try do
+              job |> Job.decode! |> start_job(state)
```

Comment:
Would it be preferable to have `Job.decode!` not raise, and instead return `{:ok, job}`, `{:error reason}` tuple or at least rescue on a specific exception type?